### ISSUE TITLE
want ability to pass parent info explicitly to Tracer.Start

### ISSUE THREAD

**Issue by andreimatei** [open-telemetry/opentelemetry-go#2251]
### Problem Statement

I would like to be able to pass the parent `Span/SpanContext` explicitly to `tracer.Start(ctx, ...)`, not through `ctx`. This is because  I have a use case where the parent span is not naturally part of `ctx`, so the current API is forcing me to allocate a `ctx` just so that I can put the parent info in it.

### Proposed Solution

I'd like new `SpanOptions`:
- `WithParent(Span)`
- `WithParentContext(SpanContext)` (or perhaps `WithRemoteParent`)

#### Prior Art

In the `opentracing` API, the parent was not inferred from the `ctx`. The API has a `SpanReference` option.

### Additional Context

In CockroachDB, we have fairly sophisticated tracing support. We have our own tracer, which can optionally tee off to an otel `Tracer`. The otel information is not put in the `Context` directly; it is embedded in our own span types. In addition, we're pretty dynamic-allocation sensitive, and the `context.Context` API is very allocation-happy. So it'd be good if we had a way to be as explicit as possible when creating otel spans.

**Comment by Aneurysm9** [comment]
The [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#span-creation) does not allow for this and explicitly prohibits it:

> The API MUST accept the following parameters:
> ...
> *  The parent `Context` or an indication that the new `Span` should be a root `Span`. The API MAY also have an option for implicitly using the current `Context` as parent as a default behavior. This API MUST NOT accept a `Span` or `SpanContext` as parent, only a full `Context`.

The spec would need to change before we could consider this request.

**Comment by andreimatei** [comment]
I don't have any context on the spec (ha!) and the text doesn't go into its thinking, but this seems like a weird prohibition to me. What's your take on my request? Would you advise that I open a discussion on the spec's repo?

**Comment by MrAlias** [comment]
> Would you advise that I open a discussion on the spec's repo?

Yes. This will need to be resolved there before we can address it here.

**Comment by andreimatei** [comment]
Thanks. I'll take this to mean that y'all would be opened to a patch if the spec was changed. I've opened https://github.com/open-telemetry/opentelemetry-specification/issues/1943. I'll see how it goes and come back.
